qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -112,3 +122,39 @@ def encode(self, op: QuadraticProgram, penalty_factor: float = 1e5,
             self._dst.maximize(offset, linear, quadratic)
 
         return self._dst
+
+    def _auto_define_penalty(self, penalty_factor: float = 1e5) -> float:
+        """Automatically define the penalty coefficient.
+
+        This returns (upper bound - lower bound + 1) of the objective function.
+
+        Args:
+            penalty_factor: The penalty factor from `LinearEqualityToPenalty.encode`
+
+        Returns:
+            Return the minimum valid penalty factor calculated
+            from the upper bound and the lower bound of the objective function
+        """
+
+        # if a constraint has a float coefficient,
+        # return input `penalty_factor` for the penalty factor.
+        terms = []
+        for constraint in self._src.linear_constraints:
+            terms.append(constraint.rhs)
+            terms.extend(coef for coef in constraint.linear.to_dict().values())
+        if any(isinstance(term, float) and not term.is_integer() for term in terms):
+            logger.warning('Using %f for the penalty coefficient because '
```

Comment:
Can we unit test this exception path to make sure its working as expected? Currently this path that logs the warning and returns what was passed in is an untested path.